cupboard_app/test_queries.py

```diff
@@ -88,6 +92,72 @@ def test_get_ingredient(self):
             get_ingredient(name='doesnt_exist')
 
 
+class CustomIngredientQueries(TestCase):
+    ing1 = None
+    ing2 = None
+    user = None
+
+    def setUp(self):
+        self.user = User.objects.create(username='test_user', email='test_user@cupboard.app')
+        self.ing1 = CustomIngredient.objects.create(
+            user=self.user, name='test_ingredient1', type='test_type1'
+        )
+        self.ing2 = CustomIngredient.objects.create(
+            user=self.user, name='test_ingredient2', type='test_type1'
+        )
+
+    def test_create_custom_ingredient(self):
+        """
+        Testing create_custom_ingredient creates a custom ingredient for the user
+        """
+        create_custom_ingredient(username='test_user', name='test_ingredient3', type='test_type2')
+        self.assertEqual(
+            CustomIngredient.objects.filter(
+                user=self.user, name='test_ingredient3', type='test_type2'
+            ).exists(),
+            True
+        )
+        self.assertEqual(len(CustomIngredient.objects.all()), 3)
+
+        create_custom_ingredient(username='test_user', name='test_ingredient3', type='test_type2')
+        self.assertEqual(len(CustomIngredient.objects.all()), 3)
+
+    def test_get_all_custom_ingredients(self):
+        """
+        Testing get_all_custom_ingredients retrieves all the custom ingredients for the user
+        """
+        user2 = User.objects.create(username='test_user2', email='test_user2@cupboard.app')
+        CustomIngredient.objects.create(user=user2, name='test_ingredient3', type='test_type1')
+        ingredients_list = get_all_custom_ingredients(self.user.username)
+        self.assertEqual(len(ingredients_list), 2)
+        self.assertEqual(
+            json.dumps({'name': self.ing1.name, 'type': self.ing1.type}),
+            str(ingredients_list[0])
+        )
+        self.assertEqual(
+            json.dumps({'name': self.ing2.name, 'type': self.ing2.type}),
+            str(ingredients_list[1])
+        )
+
+    def test_get_custom_ingredient(self):
+        """
+        Testing get_ingredient returns an ingredient from the database
+        """
+        test_ingredient = get_custom_ingredient(
+            username=self.user.username,
+            name=self.ing1.name
+        )
+        test_ingredient2 = get_custom_ingredient(
+            username=self.user.username,
+            name=self.ing2.name, id=self.ing2.id
+        )
+        self.assertEqual(test_ingredient, self.ing1)
+        self.assertEqual(test_ingredient2, self.ing2)
+
+        with self.assertRaises(CustomIngredient.DoesNotExist):
+            get_custom_ingredient(username='test_user', name='doesnt_exist')
```

Comment:
Change the username to do `username=self.user.username` so that we have less hardcoded strings we have to change if we need to edit in the future.

Reply:
done 👍